test/unittests/libp2p/net.cpp

```diff
@@ -56,6 +56,33 @@ class TestHost: public Worker
     ba::io_service m_io;
 };
 
+template <typename T>
+class ConcurrentQueue
```

Comment:
Have you seen https://github.com/ethereum/aleth/blob/master/libdevcore/concurrent_queue.h ?

Reply:
Oh I haven't! Looks very similar, I'll remove mine then.

Where is it used though?

Reply:
Seems nowhere currently. I was using it for SmartVM previously to JIT contracts in background thread. You can adjust it to your needs.